.github/workflows/dependent-issues.yml

```diff
@@ -0,0 +1,28 @@
+name: Dependent Issues
+
+on:
+  issues:
+  pull_request:
```

Comment:
@z0al I think the pull_request event is the problem here. Maybe removing that would simply work? Let's check whether the action which is triggered using cron job actually work. Because iirc, pull_requests events GitHub token has no write access while the same thing run through cron event has write permission.

Reply:
I'm not sure I understand.

> I think the pull_request event is the problem here. Maybe removing that would simply work? because iirc, pull_requests events GitHub token has no write access while the same thing run through cron event has write permission.

IMO, it shouldn't make a difference, the action should fail the moment it tries to write anything (adding/removing a label or comment, or setting the PR status). Do you have a link where that part is mentioned?

Anyway, `pull_request` and `issues` here are not strictly mandatory for this action to work. when the action is triggered by a pull request or an issue the action would receive a payload with the target PR/issue. In that case the action will simply only run checks for that issue. Otherwise, (i.e. no issue, same as Cron job) it will check all open issues/PRs.

> Let's check whether the action which is triggered using cron job actually work

That shouldn't make any difference as stated above.

Here is the relevant part from the docs: 

> GitHub automatically creates a GITHUB_TOKEN secret to use in your workflow. You can use the GITHUB_TOKEN to authenticate in a workflow run.
> 
> When you enable GitHub Actions, GitHub installs a GitHub App on your repository. The GITHUB_TOKEN secret is a GitHub App installation access token. You can use the installation access token to authenticate on behalf of the GitHub App installed on your repository. The token's permissions are limited to the repository that contains your workflow. For more information, see "Permissions for the GITHUB_TOKEN."
> 
> Before each job begins, GitHub fetches an installation access token for the job. The token expires when the job is finished.

The issue here is that when running a forked repository the generated token will be set to `read-only` and you have to set a different token yourself.

> Also personal access tokens are also not passed in the pull request event so that also may not work.

Again, the token has nothing to do with the actual event. If you created a secret in [Settings > Secret](https://github.com/ManimCommunity/manim/settings/secrets/actions) e.g. `MY_TOKEN` you can use in the workflow as `${{ secrets.MY_TOKEN }}` and should always be passed. Tools like [semantic-release](https://github.com/semantic-release/semantic-release#readme) will use that to set a custom `GH_TOKEN` since the default doesn't have the necessary permissions.

Reply:
> Anyway, pull_request and issues here are not strictly mandatory for this action to work.

To clarify, there are there to speed up the flow. Imagine opening a PR and waiting 24 hours for the action to run and check the dependencies.

Reply:
> Do you have a link where that part is mentioned?

https://docs.github.com/en/free-pro-team@latest/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories that's what I infer from the docs.

> That shouldn't make any difference as stated above.

I would rather wait and see for another 8 hours 😉 

> To clarify, these are there to speed up the flow. Imagine opening a PR and waiting 24 hours for the action to run and check the dependencies

I think we can make the action run more frequently, maybe once every 4 hours?

Reply:
Let's focus on one thing at a time. I agree with @z0al that this should run as soon as a PR is open. The question of modifying the update frequency after that can be fixed later.

Reply:
One thing we need to check whether this actually works. If my guesses are right, the one which runs daily should work, if it didn't I don't know what to do. But if it did work, then the running it for all pull requests is not a possible one.

Reply:
FYI!, https://github.com/ManimCommunity/manim/actions/runs/392723135 passed, so what I guess is correct. @z0al

Reply:
Maybe, or maybe not :wink:  . 

I just pushed a fix for a silly bug that caused the action to exit early. So in the case of the scheduled run above, it's most likely the reason why it succeeded. The action used to exit after the first issue (which happened not to have dependencies in our case i.e. nothing to update/write).

Can we schedule the action run for something shorter? e.g. every 30mins (`*/30 * * * *`)

Reply:
> I just pushed a fix for a silly bug that caused the action to exit early. So in the case of the scheduled run above, it's most likely the reason why it succeeded. The action used to exit after the first issue (which happened not to have dependencies in our case i.e. nothing to update/write).

I am not exactly a greater expert here. So, I can't exactly tell about it.

> Can we schedule the action run for something shorter? e.g. every 30mins (*/30 * * * *)

Yeah, you can make a new PR?

Reply:
Thanks @z0al for all the help! We are now getting a new error:
```
Run z0al/dependent-issues@v1
  with:
    label: dependent
    check_issues: on
    keywords: depends on, blocked by
  env:
    GITHUB_TOKEN: ***
Context
Checking #797
  Error: Resource not accessible by integration
```

Reply:
Perhaps this is related to the access token permissions you mentioned before? (I think I forgot to create a new token :eyes: :sweat_smile: )

Reply:
That's the same error @leotrs . The `Context ...etc` part is just some logs I added. Looks like the action is trying to update the status of the PR with that number and basically failed to do so due to permission errors.

Please create a Personal access token as described in [this comment](https://github.com/ManimCommunity/manim/pull/777#issuecomment-735765972) and use it in the workflow instead of `GITHUB_TOKEN`.

Reply:
Note that you can't create a secret as `GITHUB_TOKEN`. Maybe name it something like `GH_TOKEN`

Reply:
:+1:

Reply:
I still wonder whether Github action allows the passing of secrets to pull requests? Does it?